x-pack/plugins/reporting/public/plugin.ts

```diff
@@ -249,9 +260,12 @@ export class ReportingPublicPlugin
               toasts,
               uiSettings,
               license,
-              application,
+              application: core.application,
               usesUiCapabilities,
               theme: core.theme,
+              overlays: core.overlays,
+              i18nStart: core.i18n,
+              urlService: urlSetup,
```

Comment:
Maybe we can clean up the registrations by destructuring what we want from coreStart, and using a common "opts" object, like:
```typescript
    startServices$.subscribe(([coreStart, { licensing }]) => {
      const { application, overlays, i18n: i18nStart } = coreStart;

      licensing.license$.subscribe((license) => {
        const shareProviderOpts = {
          apiClient,
          toasts,
          uiSettings,
          license,
          application,
          usesUiCapabilities,
          theme: core.theme,
          overlays,
          i18nStart,
          urlService: urlSetup,
        };
        shareSetup.register(reportingCsvShareProvider(shareProviderOpts));

        if (this.config.export_types.pdf.enabled || this.config.export_types.png.enabled) {
          shareSetup.register(reportingScreenshotShareProvider(shareProviderOpts));
        }
      });
    });
```

Reply:
clean! I like it thanks!